### PR TITLE
Update splitListItem.ts to support overrideAttrs

### PR DESCRIPTION
**Pull request by Nantris**
## Please describe your changes

This change addresses #4182

## How did you accomplish your changes

Use object spread to allow `overrideAttrs` to override the default determined attributes

## How have you tested your changes

_Very minimally!_ I only tested these changes via `patch-package` and then free-handed them in using the online Github editor

## How can we verify your changes

I have not added any relevant tests.

## Remarks

I have not updated the docs

## Checklist

- [x] The changes are not breaking the editor
- [ ] Added tests where possible
- [x] Followed the guidelines
- [x] Fixed linting issues

## Related issues

#4182

### CODE REVIEW

**Comment by Slapbox** [comment]
This PR needs additional work to be merge-worthy. I hope someone will pickup and complete the remaining tasks as I will not have the time.

File: packages/core/src/commands/splitListItem.ts [hunk 1]
```diff
@@ -68,11 +68,11 @@ export const splitListItem: RawCommands['splitListItem'] = typeOrName => ({
         const depthAfter = $from.indexAfter(-1) < $from.node(-2).childCount ? 1 : $from.indexAfter(-2) < $from.node(-3).childCount ? 2 : 3
 
       // Add a second list item with an empty default start node
-      const newNextTypeAttributes = getSplittedAttributes(
+      const newNextTypeAttributes = {...getSplittedAttributes(
```
Comment (bdbch) [inline packages/core/src/commands/splitListItem.ts:71]: We have some linting issues in here @Slapbox, could we get those resolved?

File: packages/core/src/commands/splitListItem.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ declare module '@tiptap/core' {
   }
 }
 
-export const splitListItem: RawCommands['splitListItem'] = typeOrName => ({
+export const splitListItem: RawCommands['splitListItem'] = (typeOrName, overrideAttrs = {}) => ({
```
Comment (bdbch) [inline packages/core/src/commands/splitListItem.ts:22]: Maybe I'm a bit blind but won't Typescript define the type for `overrideAttrs` to an empty object? Don't you need to add a type for `overrideAttrs` here?

https://github.com/ueberdosis/tiptap/blob/fc62eb3c4d6d784b5f9b43a58e0bb0a1073cb612/packages/core/src/commands/splitListItem.ts#L17C23-L17C23

I'd suggest just to set it to `[key: string]: any`?

**Review by bdbch — CHANGES_REQUESTED** [review]
Linting and Typing issues I guess? Otherwise this looks fine, thank you.

File: packages/core/src/commands/splitListItem.ts (hunk 2 above)
Comment (Slapbox) [reply packages/core/src/commands/splitListItem.ts:22]: Good catch!

**Comment by Slapbox** [comment]
I made these edits in a proper editor with ESLint support this time, so I think they should pass the tests.

**Comment by Slapbox** [comment]
Although this commit still needs the associated docs update.

**Review by bdbch — APPROVED** [review]

**Comment by bdbch** [comment]
LGTM for now – feel free to add docs to the PR

**Review by nperez0111 — APPROVED** [review]

**Pull request merged**